source/module_esolver/esolver_ks_lcao_elec.cpp

```diff
@@ -716,6 +716,9 @@ void ESolver_KS_LCAO<TK, TR>::nscf(void)
         GlobalC::ld.cal_gedm(GlobalC::ucell.nat);
     }
 #endif
+
+    this->create_Output_Mat_Sparse(0).write();
```

Comment:
any new tests needed for this function?

Reply:
In SCF calculation, the output of Hamiltonian, Overlap, rR matrix has been supported and tested.
But in NSCF calculation, these files won't be printed with input parameters, which looks like a bug.
Some users want to skip SCF calculation and output these files.
New tests for this feature is necessary, I will modify existed test to cover this feature without extra testing time.

Reply:
OK， then I will merge this PR after the tests have been added.

Reply:
OK